pandas/tests/frame/test_combine_concat.py

```diff
@@ -129,6 +129,17 @@ def test_concat_tuple_keys(self):
         )
         assert_frame_equal(results, expected)
 
+    def test_append_empty_list(self):
+        df = DataFrame()
+        result = df.append([])
+        expected = df.copy()
```

Comment:
I don't think the copy is necessary, but since DataFrame.append should return a new object, we should probably also test that the result is not df.